lib/services/local/generic/generic.go

```diff
@@ -280,8 +282,15 @@ func (s *Service[T]) MakeBackendItem(resource T, name string) (backend.Item, err
 }
 
 // MakeKey will make a key for the service given a name.
-func (s *Service[T]) MakeKey(name string) []byte {
-	return backend.Key(s.backendPrefix, name)
+func (s *Service[T]) MakeKey(name string, parts ...string) []byte {
+	numKeyParts := len(parts)
+	keyParts := make([]string, numKeyParts+2)
+	keyParts[0] = s.backendPrefix
+	keyParts[numKeyParts+1] = name
+	if numKeyParts > 0 {
+		copy(keyParts[1:], parts)
+	}
+	return backend.Key(keyParts...)
```

Comment:
```suggestion
	return backend.Key(append([]string{s.backendPrefix, name}, parts...)...)
```

(There may be a more elegant way.)

Reply:
This has the potential to return unexpected objects when doing pagination -- I think the extra namespacing bits need to be prefixed prior to the name here.

Reply:
Could we then change the signature to `MakeKey(names ...string)` and let the caller pass everything in the correct order? Or did I misunderstand it?

Reply:
You didn't misunderstand it, but doing it that way creates an unfortunate amount of work on the caller. Basically, every caller would need to do this bit in order to get the proper ordering then:
```
	numKeyParts := len(parts)
	keyParts := make([]string, numKeyParts+2)
	keyParts[0] = s.backendPrefix
	keyParts[numKeyParts+1] = name
	if numKeyParts > 0 {
		copy(keyParts[1:], parts)
	}
        key = s.MakeKey(keyParts...)
```

I tried (a lot) to make it work the way you're describing and it ended up in a lot of duplication.

Reply:
I still think it's doable, but looking back I think the naming/ordering threw me off.

Suggestion:

```go
// MakeKey returns a key in the format [s.backendPrefix, prefixes..., suffix].
func (s *Service[T]) MakeKey(suffix string, prefixes ...string) []byte {
	l := len(prefixes)
	parts := make([]string, l+2)
	parts[0] = s.backendPrefix
	if l > 0 {
		copy(parts[1:], prefixes)
	}
	parts[l+1] = name
	return backend.Key(parts...)
}
```

(and then ditch the MakeKey test?

Reply:
I can make the function change. What's the problem with the make key test, though? It's still nice to have this test coverage IMO.

Reply:
No problem, it's just that MakeKey gets plenty of "exercise" from all other functions. If it's trivial to read, then I don't see much issue with it not getting a dedicated test.